packages/slate/src/models/text.js

```diff
@@ -88,6 +88,15 @@ class Text extends Record(DEFAULTS) {
     const { key = generateKey() } = object
     let { leaves = List() } = object
 
+    if (leaves.size === 0 && object.ranges) {
```

Comment:
I don't think this will work as written for JSON arrays, since they need `length === 0` instead.

Reply:
@ianstormtaylor that is correct! I guess we could use this instead, if you have no objections, I’ll gladly update the pull request :

```javascript
    let { leaves } = object

    if (!leaves) {
      if (object.ranges) {
        logger.deprecate(
          'slate@0.27.0',
          'The `ranges` property of Slate objects has been renamed to `leaves`.'
        )
        leaves = object.ranges
      } else {
        leaves = List()
      }
    }
```